truelearn/datasets/_base.py

```diff
@@ -0,0 +1,105 @@
+import dataclasses
+import hashlib
+from typing import Optional
+from urllib import request
+from os import path
+
+
+# pylint: disable=pointless-string-statement
+"""
+Copyright of `RemoteFileMetaData`, `download_file` are held by
+[BSD 3-Clause License, scikit-learn developers, 2007-2022].
+"""
+
+
+@dataclasses.dataclass
+class RemoteFileMetaData:
+    """Remote file metadata.
+
+    Args:
+        url: The url of the file.
+        filename: The filename of the downloaded file.
+        sha256_expected: The expected sha256 sum of the file.
+    """
+
+    url: str
+    filename: str
+    sha256: str
+
+
+def _sha256sum(filepath) -> str:
+    buf_size = 65536
```

Comment:
What is the reason behind this buffer size?

Reply:
I chose it because it is not too big/small. If the buffer size is too large, the function will consume too much memory. Conversely, if the buffer size is too small, it may take longer to complete.

So, I choose to use `65536` as the buffer size (Note, the best buffer size needs to be chosen based on experimentation. However, the PEEKdataset is not very large. So, I didn't take the time to test the "right" buffer size, and I suspect it won't make much difference). Please refer to [this question](https://stackoverflow.com/questions/22058048/hashing-a-file-in-python) for more details.